### PR TITLE
Andrew

### LINKED ISSUE #37: Enforce DataFrame formatting
Or, add tests in the degradations with other formats.

For example, there are currently no tests with dataframes without consecutive indices starting from 0, or unsorted dataframes, or ones with columns out of order.

Some of these we might want to allow, some we might be okay with errors.

We could also add a function which, given some data frame, enforces formatting, like:
1. Drops all columns not pitch, onset, offset, track
2. Rounds those columns to ints
3. Puts those columns in the correct order
4. Sorts the dataframe
5. Reset the index in that order and drop the index column

Currently, the degradations are closed under 1, 2, 3, and 5 (if given correctly, will return correctly), but not necessarily 4 (the resulting df may not be sorted, even if the input one was).

Giving one a df which has the wrong columns would likely error.

### PR DESCRIPTION
**Pull request by apmcleod**
This fixes #37. I confirmed that the degs do error with incorrect input, and added a nice error message.

This also does a lot of work on #132. Basically, it naively changes all warnings.warn to logging.warning. Functionality will not change, but we may want to do some filtering or something with the logging module before closing the issue.

### CODE REVIEW

File: make_dataset.py [hunk 1]
```diff
@@ -305,10 +297,8 @@ def parse_args(args_input=None):
     # Warn user they specified kwargs for degradation not being used
     for deg, args in degradation_kwargs.items():
         if deg not in ARGS.degradations and len(args) > 0:
-            warnings.warn(
-                "--degradation_kwargs contains args for unused "
-                f'degradation "{deg}".',
-                UserWarning,
+            logging.warning(
+                "--degradation_kwargs contains args for unused " f'degradation "{deg}".'
```
Comment (JamesOwers) [inline make_dataset.py:301]: Can just be one f-string as opposed to two adjacent

**Review by JamesOwers — APPROVED** [review]
Looks good to me bar one optional string format change.

I wonder why we wrote the groupby in two parts before? There is likely some faster vectorised method we could use in future, but for now this is fine. It's easy to read and that is good!

File: make_dataset.py (hunk 1 above)
Comment (apmcleod) [reply make_dataset.py]: Done. Also fixed where this happened elsewhere due to the auto-formatter.

**Comment by apmcleod** [comment]
It was a nested groupby before because I didn't know you could group by multiple columns at the same time :P

**Pull request merged**